docs/book/src/developer/providers/migrations/v1.10-to-v1.11.md

```diff
@@ -3,81 +3,266 @@
-  - All the resources handling machine replicas have now a consistent set of replica counters based on corresponding
-    conditions defined at machine level.
-    - `status.readyReplicas`, `status.availableReplicas`, `status.upToDateReplicas` on `MachineDeployments`, `MachineSet`
-      and `KubeadmControlPlane`; please note that
-      - `status.readyReplicas` has now a new semantic based on machine's `Ready` condition
-      - `status.availableReplicas` has now a new semantic based on machine's `Available` condition
-      - `status.upToDateReplicas` has now a new semantic (and name) based on machine's `UpToDate` condition
-      - Temporarily, old replica counters will still be available under the `status.deprecated.v1beta1` struct; more specifically
-        - `status.deprecated.v1beta1.readyReplicas` with old semantic for `MachineDeployments`, `MachineSet` and `KubeadmControlPlane`
-        - `status.deprecated.v1beta1.availableReplicas` with old semantic for `MachineDeployments`, `MachineSet`
-        - `status.deprecated.v1beta1.unavailableReplicas` with old semantic for `MachineDeployments`, `KubeadmControlPlane`
-        - `status.deprecated.v1beta1.updatedReplicas` with old semantic (and name) for `MachineDeployments`, `KubeadmControlPlane`
-        - `status.deprecated.v1beta1.fullyLabeledReplicas` for `MachineSet`
-    - The `Cluster` resource reports replica counters for both control plane and worker machines.
-  - MachineDeployment `status.phases` are now computed using the same logic used for `ScalingUp` and `ScalingDown` conditions.
+- v1beta2 API version has been introduced; see following paragraphs for more details
+- If you are using the `runtime.cluster.x-k8s.io` API group, please be aware that
+  - `ExtensionConfig` v1alpha1 has been deprecated, and it will be removed in a following release.
+  - `ExtensionConfig` v1beta2 has created (thus aligning with other Cluster API resources)
+
+ ### All CRDs
+
+- The transition to the new K8s aligned conditions using `metav1.Conditions` types and the new condition semantic
+  has been completed for all Kinds:
+  - `status.conditions` has been replaced with `status.v1beta2.conditions` based on metav1 condition types
+  - the old `status.conditions` based on custom cluster API condition types will continue to exist temporarily 
+    under `status.deprecated.v1beta1.conditions` for the sake of down conversions and to provide a temporary option
+    for users willing to continue using old conditions.
+
+### Cluster
+
+- See changes that apply to [all CRDs](#all-crds)
+- The `spec.topology.class` field has been renamed to `spec.topology.classRef.name`
+- The `spec.topology.classNamespace` field has been renamed to `spec.topology.classRef.namespace`
+- The `definitionFrom` field (deprecated since CAPI v1.8) has been removed from
+  - `spec.topology.variables`
+  - `spec.topology.controlPlane.variables.overrides`
+  - `spec.topology.workers[].machineDeployments.variables.overrides`
```

Comment:
missed before. There are a few cases where the array is on the wrong level (machineDeployments is the array not workers)